src/record/mod.rs

```diff
@@ -89,9 +89,15 @@ pub trait Record: 'static + Sized + Decode + Debug + Send + Sync {
         self.as_record_ref().key()
     }
 
+    /// Returns owned value of record (self)
+    fn as_owned_value(&self) -> Self;
+
     /// Returns a reference to the record.
     fn as_record_ref(&self) -> Self::Ref<'_>;
 
+    /// Applies projection mask
+    fn projection(&mut self, projection_mask: &ProjectionMask);
```

Comment:
unifying projection/limit/filter behind a small logical expression is a good idea instead.